pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1504,24 +1521,38 @@ func (volTopology *wcpControllerVolumeTopology) GetSharedDatastoresInTopology(ct
 		// Call GetCandidateDatastores for each cluster moref. Ignore the vsanDirectDatastores for now.
 		if !isPodVMOnStretchedSupervisorEnabled {
 			// This code block assume we have 1 Cluster Per AZ
-			accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMorefs[0], false)
+			getCandidateFunc := volTopology.getCandidateDatastoresFunc
+			if getCandidateFunc == nil {
+				getCandidateFunc = cnsvsphere.GetCandidateDatastoresInCluster
```

Comment:
since we are setting getCandidateFunc = volTopology.getCandidateDatastoresFunc, when do we set getCandidateFunc to cnsvsphere.GetCandidateDatastoresInCluster, as getCandidateFunc == nil will be always false.

Reply:
I made some changes so it does not do nil check any more.